pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -105,24 +107,32 @@ func NewDiscoveryServer(env *model.Environment, generator core.ConfigGenerator)
 
 	out.DebugConfigs = os.Getenv("PILOT_DEBUG_ADSZ_CONFIG") == "1"
 
-	pushThrottle := intEnv("PILOT_PUSH_THROTTLE", 10)
-	pushBurst := intEnv("PILOT_PUSH_BURST", 100)
+	pushThrottle := intEnv("PILOT_PUSH_THROTTLE", 25)
+	pushBurst := intEnv("PILOT_PUSH_BURST", 200)
 
 	adsLog.Infof("Starting ADS server with throttle=%d burst=%d", pushThrottle, pushBurst)
-	rate := time.Second / time.Duration(pushThrottle)
-	burstLimit := pushBurst
-	tick := time.NewTicker(rate)
-	out.throttle = make(chan time.Time, burstLimit)
+
+	// init throttle affects new connections from sidecars.
+	out.initThrottle = initThrottle("initConnection", pushBurst, pushThrottle*2)
+	// normal throttle is set to half of
+	out.throttle = initThrottle("adsPushAll", pushBurst, pushThrottle)
+
+	return out
+}
+
+// initThrottle allocates and initializes a throttle channel with burstLimit and steady state ratePerSecond.
+func initThrottle(name string, burst int, ratePerSecond int) chan time.Time {
+	tick := time.NewTicker(time.Second / time.Duration(ratePerSecond))
+	throttle := make(chan time.Time, burst)
 	go func() {
 		for t := range tick.C {
 			select {
-			case out.throttle <- t:
+			case throttle <- t:
```

Comment:
how is this going to add burst? this is simply adding an event every t seconds.. and you have created a backlog size of len==burst. But thats not what burst is.

Reply:
initially the channel starts empty and tokens are added at the specified rate, to a limit of `burst`.
This means that at any time the maximum number of token that can be consumed at one time is 100, so we allow bursts of up to 100 requests instantaneously, and then only allow more work at 25/s as specified.